extensions/fetch/26_fetch.js

```diff
@@ -429,6 +431,17 @@
       }
       requestObject.signal[abortSignal.add](onabort);
 
+      const baseURL = getLocationHref();
+      if (
+        baseURL &&
+        (requestObject.method !== "GET" && requestObject.method !== "HEAD")
+      ) {
+        ArrayPrototypePush(request.headerList, [
+          "origin",
+          new URL(baseURL).origin,
+        ]);
+      }
+
```

Comment:
This is not correct. The header should be added as per of a different stage of the implementation.

The origin header should also respect request referrer policy.

Reply:
this section? https://fetch.spec.whatwg.org/#append-a-request-origin-header

Reply:
Yeah, take a look at where it is called.